src/validation.cpp

```diff
@@ -3954,17 +3954,7 @@ bool LoadChainTip(const CChainParams& chainparams)
 
     if (::ChainActive().Tip() && ::ChainActive().Tip()->GetBlockHash() == pcoinsTip->GetBestBlock()) return true;
 
-    if (pcoinsTip->GetBestBlock().IsNull() && mapBlockIndex.size() == 1) {
-        // In case we just added the genesis block, connect it now, so
-        // that we always have a ::ChainActive().Tip() when we return.
-        LogPrintf("%s: Connecting genesis block...\n", __func__);
-        CValidationState state;
-        if (!ActivateBestChain(state, chainparams)) {
-            LogPrintf("%s: failed to activate chain (%s)\n", __func__, FormatStateMessage(state));
-            return false;
-        }
-    }
-
+    assert(!pcoinsTip->GetBestBlock().IsNull());
```

Comment:
Could move this up, before above condition?

Reply:
Done and added comment

Reply:
Cool.